engine/crates/composition/src/compose.rs

```diff
@@ -100,6 +101,10 @@ fn merge_object_definitions<'a>(
         ctx.insert_resolvable_key(object_id, key.id);
     }
 
+    if is_shareable {
```

Comment:
This is not the complete solution: we should check shareable fields inside non-shareable objects, take into accounts provides and requires, etc. It gets very complicated, but this is an 80% solution as a first step.